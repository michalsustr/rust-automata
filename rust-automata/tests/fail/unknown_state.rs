use rust_automata::*;

#[derive(Default)]
pub struct S1;
#[derive(Default)]
pub struct I1;

#[automaton(
    inputs(I1),
    states(S1),
    outputs(),
    initial_state(S1),
    transitions(
        (S1, I1) -> (S2)   // S2 is not declared
    )
)]
pub struct BadState;

fn main() {}
