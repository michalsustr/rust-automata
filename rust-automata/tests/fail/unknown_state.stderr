error: cannot find attribute `automaton` in this scope
 --> tests/fail/unknown_state.rs:8:3
  |
8 | #[automaton(
  |   ^^^^^^^^^

warning: unused import: `rust_automata::*`
 --> tests/fail/unknown_state.rs:1:5
  |
1 | use rust_automata::*;
  |     ^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
